Add compat data for ::slotted pseudo-element

chrisdavidmills opened the pull request.

teoli2003 commented:
This looks like a recent addition. @chrisdavidmills: Do you have a bug number so I can check the content of the table, and if is behind a flag?

chrisdavidmills commented:
This is a good point — serves me right for doing a PR so late at night ;-)

The ::slotted() meta bug seems to be this one: https://bugzilla.mozilla.org/show_bug.cgi?id=1424604

And this one suggests that it'll only be parsed if the webcomponents pref is on: https://bugzilla.mozilla.org/show_bug.cgi?id=1425834

Ummm, there are three ;-)

No matter. I have tested my demo, and it only parses ::slotted if dom.webcomponents.customelements.enabled and dom.webcomponents.shadowdom.enabled are set to true.

So I guess I need to add some notes to Firefox/FxA.

I am not sure about Chrome/Opera versions. I had difficulty digging up any more exact bugs, but I know that it works in both Chrome 63 and Opera 50 (release versions).

teoli2003 commented:
Thanks, Chris!

@jpmedley  may help us with Chromia :-)

teoli2003 commented:
Also, I wonder if it does anything that is actually usable on Firefox right now, or if it only an implementation in progress, and we should keep at false (not implemented) until the implementation is finished. @emilio, as the implementer, what do you think?

jpmedley commented:
Chrome 50/Opera 37

chrisdavidmills commented:
Thanks folks. I've updated the info with the new details.

emilio commented:
@chrisdavidmills it should be usable on nightly, but it's preffed-off (so not shipped). So given it's not usable on release I'd maybe punt on it.

teoli2003 commented on css/selectors/slotted.json:
```diff
@@ -0,0 +1,79 @@
+              "version_added": null
+            },
+            "firefox": {
+              "version_added": true,
```
"59"

teoli2003 commented on css/selectors/slotted.json:
```diff
@@ -0,0 +1,79 @@
+              ]
+            },
+            "firefox_android": {
+              "version_added": true,
```
"59"

chrisdavidmills commented:
I've updated the data as requested; I've also added an entry for it to the experimental features page:

https://developer.mozilla.org/en-US/Firefox/Experimental_features#CSS

@emilio I think we should include it now; quite a few devs are interested in how they can enable and play with new features, even if they aren't useful for users yet.

emilio commented:
Ok, sounds good then! :)
